doc/source/dev/api_guidelines.rst

```diff
@@ -0,0 +1,172 @@
+API Design Guidelines
+=====================
+
+The following section outlines guidelines related to extending and/or modifying
+the Galaxy API. The Galaxy API has grown in an ad-hoc fashion over time by
+many contributors and so clients SHOULD NOT expect the API will conform to
+these guidelines - but developers contributing to the Galaxy API SHOULD follow
+these guidelines.
+
+Before Getting Started
+~~~~~~~~~~~~~~~~~~~~~~
+
+Developers should familiarize themselves with HTTP method definitions 
+https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9. While Galaxy
+follows these imperfectly - a serious effort should be made to use HTTP
+methods (GET, POST, PUT, DELETE, etc.) correctly and consistently. More
+information about these methods with a focus on RESTful APIs can be found
+at https://restfulapi.net/http-methods/.
```

Comment:
```suggestion
at https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design#conform-to-http-semantics .
```
(As per discussion on resgulapi.net) + this part of Microsoft's guide is, I think, exactly "...these methods with a focus on RESTful APIs"

Reply:
I didn't catch that was still there - thanks for catching that.